Shopify/shopify-cli
# Feature: Unconditionally enable new extension flow for Checkout Extensions

### WHY are these changes introduced?

[Enable Development Server for Checkout Extensions](https://github.com/Shopify/shopify-cli-extensions/issues/238). This PR has a [dependency](https://github.com/Shopify/shopify-cli/pull/2131).

### WHAT is this pull request doing?

Enables the development server for checkout extensions regardless of the state of the "beta enabled" flag. Checkout post purchase and product subscription types are still supported and continue to be guarded by the same flag.

https://user-images.githubusercontent.com/4490738/157730914-adc0aedf-e293-4f1b-b4c0-da70684cd764.mp4

### How to test your changes?

1. Navigate to `Shopify/shopify-cli` directory.
2. Checkout this branch: `git fetch; git checkout origin/checkout-extensions-enabled`
3. Disable the beta flag with `shopify config feature extension_server_beta --disable`
4. Create a checkout extension with `bin/shopify extension create`
5. Navigate to the new extension subdirectory created with the name you provided (`cd <checkout-ext-name>`)
6. Serve the extension with `../bin/shopify extension serve extension.config.yml`
7. Going to the public URL displayed to the console should display "CHECKOUT_UI_EXTENSION" and a green "Connected" banner

(See the attached video for more detail)

### Post-release steps

### Update checklist
